static/app/components/searchSyntax/grammar.pegjs

```diff
@@ -339,8 +339,8 @@ ms_format   = [0-9] [0-9]? [0-9]? [0-9]? [0-9]? [0-9]?
 tz_format   = [+-] num2 ":" num2
 
 iso_8601_date_format
-  = date_format time_format? ("Z" / tz_format)? &end_value {
-      return tc.tokenValueIso8601Date(text());
+  = date:date_format time:time_format? tz:("Z" / tz_format)? &end_value {
```

Comment:
Adding names to these parameters so that we can put them into the token data object to be used in the query builder.

Reply:
I'm actually going to pull this out into a separate PR to make this diff a bit more bearable